### PR TITLE
NMS-14469: Add a double value wrapper to the NumericAttribute message

### PR DESCRIPTION
**Pull request by amay94**
Protobuf cannot distinguish between an unset and a default scalar value, see https://developers.google.com/protocol-buffers/docs/proto3#default

To get around that, this adds a new `DoubleValue metric_value` to the `NumericAttribute` message that is a copy of the scalar double value that can distinguish between default and unset.

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14469

### CODE REVIEW

File: features/kafka/producer/src/test/java/org/opennms/features/kafka/producer/collection/KafkaPersisterIT.java [hunk 1]
```diff
@@ -168,19 +168,42 @@ public void testKafkaCollection() throws IOException {
...
+        LatencyCollectionResource latencyCollectionResource = new LatencyCollectionResource("ICMP", IP_ADDRESS, LOCATION);
+        LatencyCollectionAttributeType attributeType = new LatencyCollectionAttributeType("ICMP", "ICMP");
+        latencyCollectionResource.addAttribute(new LatencyCollectionAttribute(latencyCollectionResource, attributeType, "ICMP", 0.0));
+        CollectionSet responseTimeCollectionSet = new SingleResourceCollectionSet(latencyCollectionResource, new Date());
+        persister.visitCollectionSet(responseTimeCollectionSet);
+
+        await().atMost(1, TimeUnit.MINUTES).pollInterval(15, TimeUnit.SECONDS).until(() -> kafkaConsumer.getCollectionSetValues().size(), equalTo(1));
+        List<CollectionSetProtos.CollectionSetResource> resources = kafkaConsumer.getCollectionSetValues().stream().map(CollectionSetProtos.CollectionSet::getResourceList).flatMap(Collection::stream).collect(Collectors.toList());
+        Optional<CollectionSetProtos.CollectionSetResource> responseTimeResource = resources.stream().filter(CollectionSetProtos.CollectionSetResource::hasResponse).findFirst();
+        responseTimeResource.ifPresent(resource -> {
```
Comment (cgorantla) [inline features/kafka/producer/src/test/java/org/opennms/features/kafka/producer/collection/KafkaPersisterIT.java:198]: should validate presence of resource and then validate the resource fields

File: features/kafka/producer/src/test/java/org/opennms/features/kafka/producer/collection/KafkaPersisterIT.java [hunk 2]
```diff
@@ -168,19 +168,42 @@ public void testKafkaCollection() throws IOException {
...
+                    assertThat(resource.getResponse().getInstance(), equalTo(IP_ADDRESS));
+                    assertThat(resource.getResponse().getLocation(), equalTo(LOCATION));
+                    assertThat(resource.getNumeric(0).getValue(), equalTo(0.0));
```
Comment (cgorantla) [inline features/kafka/producer/src/test/java/org/opennms/features/kafka/producer/collection/KafkaPersisterIT.java:201]: Is it possible to validate  that `value` is not present but it is returning default value.
may be using `hasField` ?

**Review by cgorantla — COMMENTED** [review]
lgtm,  minor comment.

May have to update docs as well.

File: features/kafka/producer/src/test/java/org/opennms/features/kafka/producer/collection/KafkaPersisterIT.java (hunk 1 above)
Comment (amay94) [reply features/kafka/producer/src/test/java/org/opennms/features/kafka/producer/collection/KafkaPersisterIT.java:208]: done

File: features/kafka/producer/src/test/java/org/opennms/features/kafka/producer/collection/KafkaPersisterIT.java (hunk 2 above)
Comment (amay94) [reply features/kafka/producer/src/test/java/org/opennms/features/kafka/producer/collection/KafkaPersisterIT.java:211]: done

**Review by cgorantla — COMMENTED** [review]
Other than adding a doc comment, lgtm!

**Comment by cgorantla** [comment]
should be re-targeted to `release-30.x`

**Comment by amay94** [comment]
https://docs.opennms.com/horizon/30/operation/kafka-producer/kafka-producer.html?q=metri#metrics

Our documentation refers to the .proto files for descriptions of the individual fields. There's a comment there, so no update to documentation needed.

**Comment by cgorantla** [comment]
> https://docs.opennms.com/horizon/30/operation/kafka-producer/kafka-producer.html?q=metri#metrics
> 
> Our documentation refers to the .proto files for descriptions of the individual fields. There's a comment there, so no update to documentation needed.

We should probably add .proto file to docs, https://issues.opennms.org/browse/NMS-13814

**Comment by amay94** [comment]
> > https://docs.opennms.com/horizon/30/operation/kafka-producer/kafka-producer.html?q=metri#metrics
> > Our documentation refers to the .proto files for descriptions of the individual fields. There's a comment there, so no update to documentation needed.
> 
> We should probably add .proto file to docs, https://issues.opennms.org/browse/NMS-13814

That's fair, but it's beyond the scope of this specific issue, right?

**Comment by cgorantla** [comment]
> > > https://docs.opennms.com/horizon/30/operation/kafka-producer/kafka-producer.html?q=metri#metrics
> > > Our documentation refers to the .proto files for descriptions of the individual fields. There's a comment there, so no update to documentation needed.
> > 
> > 
> > We should probably add .proto file to docs, https://issues.opennms.org/browse/NMS-13814
> 
> That's fair, but it's beyond the scope of this specific issue, right?

yes, you can tackle that in separate issue.

**Review by cgorantla — APPROVED** [review]

**Pull request merged**